lib/charms/hydra/v0/hydra_endpoints.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Interface library for sharing hydra endpoints.
+
+This library provides a Python API for both requesting and providing public and admin endpoints.
+
+## Getting Started
+To get started using the library, you need to fetch the library using `charmcraft`.
+```shell
+cd some-charm
+charmcraft fetch-lib charms.hydra.v0.hydra_endpoints
+```
+
+To use the library from the requirer side:
+In the `metadata.yaml` of the charm, add the following:
+```yaml
+requires:
+  endpoint-info:
+    interface: hydra_endpoints
+    limit: 1
+```
+Then, to initialise the library:
+```python
+from charms.hydra.v0.hydra_endpoints import (
+    HydraEndpointsRelationError,
+    HydraEndpointsRequirer,
+)
+
+Class SomeCharm(CharmBase):
+    def __init__(self, *args):
+        self.hydra_endpoints_relation = HydraEndpointsRequirer(self)
+        self.framework.observe(self.on.some_event_emitted, self.some_event_function)
+    def some_event_function():
+        # fetch the relation info
+        try:
+            hydra_data = self.hydra_endpoints_relation.get_hydra_endpoints()
+        except HydraEndpointsRelationError as error:
+            ...
+```
+
+"""
+
+import logging
+
+from ops.charm import (
+    CharmBase,
+    RelationChangedEvent,
+    RelationCreatedEvent,
+    RelationJoinedEvent,
+)
+from ops.framework import EventBase, EventSource, Object, ObjectEvents
+from ops.model import Application
+
+# The unique Charmhub library identifier, never change it
+LIBID = "a9dbc14576874a508dc3b7f717c72f73"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+RELATION_NAME = "endpoint-info"
+INTERFACE_NAME = "hydra_endpoints"
+logger = logging.getLogger(__name__)
+
+
+class RelationReadyEvent(EventBase):
```

Comment:
`RelationReadyEvent` is not specific enough imo, I'd replace it with `HydraEndpointsRelationReadyEvent` . Here the charm will have to use this relation and if Hydra provides multiple relation interfaces, `RelationReadyEvent` wouldn't tell us which relation is ready.